pkgs/development/python-modules/unstructured/default.nix

```diff
@@ -0,0 +1,164 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, file
+, poppler_utils
+, tesseract
+, libreoffice
+# python dependencies
+, openpyxl
+, markdown
+, python-pptx
+, msg-parser
+, pypandoc
+, argilla
+, xlrd
+, requests
+, nltk
+, certifi
+, urllib3
+, protobuf
+, jupyter-core
+, wheel
+, anyio
+, backoff
+, cffi
+, chardet
+, charset-normalizer
+, click
+, commonmark
+, cryptography
+, deprecated
+, et_xmlfile
+, exceptiongroup
+, h11
+, httpcore
+, httpx
+, idna
+, importlib-metadata
+, joblib
+, lxml
+, monotonic
+, numpy
+, olefile
+, packaging
+, pandas
+, pdfminer-six
+, pillow
+, pycparser
+, pydantic
+, pygments
+, python-dateutil
+, python-docx
+, python-magic
+, pytz
+, regex
+, rfc3986
+, rich
+, six
+, sniffio
+, tabulate
+, tqdm
+, typer
+, typing-extensions
+, wrapt
+, xlsxwriter
+, zipp
+}:
+let
+  pname = "unstructured";
+  version = "0.7.1";
+in
+buildPythonPackage {
+  inherit pname version;
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "Unstructured-IO";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    hash = "sha256-I2RUwkEKtK2NN0a5T5mEy4QpH+rdMr7z9IDO9Yy6gM8=";
+  };
+
+  buildInputs = [
+    file # libmagic
+    poppler_utils
+    tesseract
+  ] ++ lib.optionals stdenv.isLinux [ libreoffice ];
+
+  propagatedBuildInputs = [
+    openpyxl
+    markdown
+    python-pptx
+    msg-parser
+    pypandoc
+    argilla
+    xlrd
+    requests
+    nltk
+    certifi
+    urllib3
+    protobuf
+    jupyter-core
+    wheel
+    anyio
+    backoff
+    cffi
+    chardet
+    charset-normalizer
+    click
+    commonmark
+    cryptography
+    deprecated
+    et_xmlfile
+    exceptiongroup
+    h11
+    httpcore
+    httpx
+    idna
+    importlib-metadata
+    joblib
+    lxml
+    monotonic
+    numpy
+    olefile
+    packaging
+    pandas
+    pdfminer-six
+    pillow
+    pycparser
+    pydantic
+    pygments
+    python-dateutil
+    python-docx
+    python-magic
+    pytz
+    regex
+    rfc3986
+    rich
+    six
+    sniffio
+    tabulate
+    tqdm
+    typer
+    typing-extensions
+    wrapt
+    xlsxwriter
+    zipp
+  ];
```

Comment:
https://github.com/Unstructured-IO/unstructured/blob/main/requirements/base.in

Only these are propagated. The others are extras, that need to be reflected in `passthru.optional-dependencies`.

https://github.com/Unstructured-IO/unstructured/blob/main/setup.py#L72-L84